src/System.Diagnostics.Process/src/System/Diagnostics/Process.Linux.cs

```diff
@@ -0,0 +1,696 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Win32.SafeHandles;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Globalization;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading;
+
+namespace System.Diagnostics
+{
+    public partial class Process : IDisposable
+    {
+        /// <summary>
+        /// Puts a Process component in state to interact with operating system processes that run in a 
+        /// special mode by enabling the native property SeDebugPrivilege on the current thread.
+        /// </summary>
+        public static void EnterDebugMode()
+        {
+            // Nop.
+        }
+
+        /// <summary>
+        /// Takes a Process component out of the state that lets it interact with operating system processes 
+        /// that run in a special mode.
+        /// </summary>
+        public static void LeaveDebugMode()
+        {
+            // Nop.
+        }
+
+        /// <summary>Stops the associated process immediately.</summary>
+        public void Kill()
+        {
+            EnsureState(State.HaveId);
+            int errno = Interop.libc.kill(_processId, Interop.libc.Signals.SIGKILL);
```

Comment:
Do you have an feeling for if TerminateProcess (which is what we do on Windows) maps more to SIGKILL vs SIGTERM?